doc/source/data/dataset.rst

```diff
@@ -173,159 +173,6 @@ Advanced users can refer directly to the Ray Datasets :ref:`API reference <data-
         :text: Check Out Other Data Processing Options
         :classes: btn-outline-info btn-block
 
-
-.. _data-compatibility:
```

Comment:
can you check the repo and update all places refer to this link?

https://github.com/search?q=repo%3Aray-project%2Fray+data-compatibility&type=code

We can point users to https://docs.ray.io/en/master/data/api/input_output.html, instead.